src/python_testing/TC_TSTAT_4_2.py

```diff
@@ -237,126 +251,209 @@ async def test_TC_TSTAT_4_2(self):
-            status = await self.write_presets(endpoint=endpoint, presets=test_presets)
-            status_ok = (status == Status.Success)
-            asserts.assert_true(status_ok, "Presets write did not return Success as expected")
+            if availableScenario is not None and len(current_presets) < numberOfPresetsSupported:
 
-            # Read the presets attribute and verify it was updated by the write
-            presets = await self.read_single_attribute_check_success(endpoint=endpoint, cluster=cluster, attribute=cluster.Attributes.Presets)
-            logger.info(f"Rx'd Presets: {presets}")
-            asserts.assert_equal(presets, new_presets_with_handle, "Presets were updated, as expected")
+                # Set the preset builtIn fields to a null built-in value
+                test_presets = copy.deepcopy(current_presets)
+                for preset in test_presets:
+                    preset.builtIn = NullValue
 
-            await self.send_atomic_request_rollback_command()
+                test_presets.append(cluster.Structs.PresetStruct(presetHandle=NullValue, presetScenario=availableScenario,
+                                                                 coolingSetpoint=2700, heatingSetpoint=1900, builtIn=False))
+
+                await self.send_atomic_request_begin_command()
+
+                # Write to the presets attribute after calling AtomicRequest command
+                status = await self.write_presets(endpoint=endpoint, presets=test_presets)
+                status_ok = (status == Status.Success)
+                asserts.assert_true(status_ok, "Presets write did not return Success as expected")
+
+                # Read the presets attribute and verify it was updated by the write
+                saved_presets = await self.read_single_attribute_check_success(endpoint=endpoint, cluster=cluster, attribute=cluster.Attributes.Presets)
+                logger.info(f"Rx'd Presets: {saved_presets}")
+                self.check_saved_presets(test_presets, saved_presets)
 
-            # Read the presets attribute and verify it has been properly rolled back
-            presets = await self.read_single_attribute_check_success(endpoint=endpoint, cluster=cluster, attribute=cluster.Attributes.Presets)
-            asserts.assert_equal(presets, initial_presets, "Presets were updated which is not expected")
+                await self.send_atomic_request_rollback_command()
+
+                # Read the presets attribute and verify it has been properly rolled back
+                presets = await self.read_single_attribute_check_success(endpoint=endpoint, cluster=cluster, attribute=cluster.Attributes.Presets)
+                asserts.assert_equal(presets, current_presets, "Presets were updated which is not expected")
+            else:
+                logger.info(
+                    "Couldn't run test step 3 since there was no available preset scenario to append")
```

Comment:
Step 3 in the existing script appended a preset, because it saved a copy new_presets instead of a copy of initial_presets. That new preset is needed for later tests.